File: 00_AppJava2/app/src/main/java/com/example/a00_appjava2/MainActivity.java
package com.example.a00_appjava2;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Button boton1 = findViewById(R.id.boton1);
        EditText editText1 = findViewById(R.id.editText1);
        boton1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                // El objeto view que entra por aqui será
                // el objeto que ha desencadenado el evento
                Log.i("MainActivity", "Boton 1 pulsado");
            }
        });
        Button boton2 = findViewById(R.id.boton2);
        boton2.setOnClickListener(view -> Log.i("MainActivity", "Boton 2 pulsado"));
        Button boton3 = findViewById(R.id.boton3);
        ListenerBoton lb = new ListenerBoton();
        boton3.setOnClickListener(lb);

        Button boton5 = findViewById(R.id.boton5);
        TextView tx1 = findViewById(R.id.textView1);

        boton5.setOnClickListener(view -> {
            String mensaje = editText1.getText().toString();
            tx1.setText(mensaje);
        });
    }
}

File: 00_AppKotlin/app/src/main/java/com/example/a00_appkotlin/MainActivity.kt
package com.example.a00_appkotlin

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

// Para extender de las clases en Kotlin se pone :
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        var user = Usuario();
        // realmente esta invocando al setter
        user.nombre = "Guillermo"
        user.password = "1234"

        // Para invocar el metodo get usamos user.nombre y no lo igualamos
        // a nada no como arriba
        Log.d("MainActivity", "Nombre" + user.nombre)
        val boton1 = findViewById<Button>(R.id.boton1);
        val boton2 : Button = findViewById(R.id.boton2);
        boton1.setOnClickListener({
            Log.d("MainActivity", "has pulsaado el boton 1")
        })
        boton2.setOnClickListener{v->
            Log.d("MainActivity", "Has pulsado el boton 2" + v.toString())
            Log.d("MainActivity", "Es otra linea")
        }
    }
}